### PR TITLE
Add `ignoreJunk` option

### LINKED ISSUE #57: Suggestion: noJunk option
to exclude [junk](https://npmjs.com/junk) files (such as .DS_Store). This could be made available to the CLI and would reduce project dependencies.

### PR DESCRIPTION
**Pull request by whitecrownclown**
@sindresorhus 
Fixes #57 

- Added `ignoreJunk` option, defaults to `false`
- Updated readme, included tests

One thing i encountered when writing tests for this. If you glob using `*`, hidden files will be [ignored](https://github.com/mrmlnc/fast-glob#basic-syntax) by default.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,6 +47,10 @@ module.exports = (source, destination, options = {}) => {
 		let files;
 		try {
 			files = await globby(source, options);
+
+			if (options.ignoreJunk) {
+				files = files.filter(junk.not);
```
Comment (sindresorhus) [inline index.js]: `junk.not` accepts **filenames**, but `globby` produces **file paths**, which can be filenames or longer deeper paths. I think you need some more tests.

**Comment by sindresorhus** [comment]
@whitecrownclown Don't force-push. It makes it impossible for me to see what you changed, so I'm forced to re-review everything instead of just the new changes.

File: index.js [hunk 2]
```diff
@@ -46,6 +47,10 @@ module.exports = (source, destination, options = {}) => {
 		let files;
 		try {
 			files = await globby(source, options);
+
+			if (options.ignoreJunk) {
+				files = files.filter(file => junk.not(file.substring(file.lastIndexOf('/') + 1, file.length)));
```
Comment (sindresorhus) [inline index.js]: Use `path.basename()`. Your check will fail on Windows and some other situations.

File: readme.md [hunk 3]
```diff
@@ -94,6 +94,12 @@ const cpy = require('cpy');
 })();
 ```
 
+##### ignoreJunk
+
+Type: `boolean`<br>
+Default: `false`
```
Comment (sindresorhus) [inline readme.md]: Let's make it `true` by default.

**Comment by sindresorhus** [comment]
You also need to add it to https://github.com/sindresorhus/cpy/blob/master/index.d.ts

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by whitecrownclown** [comment]
I did fix the conflict, but unfortunately I messed up the rebase and had to force push.. Sorry about that.

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**